lib/system.nim

```diff
@@ -4414,6 +4414,14 @@ when not defined(js):
     magic: "Slice".}
   proc toOpenArrayByte*(x: string; first, last: int): openarray[byte] {.
     magic: "Slice".}
+  template toOpenArray*[T](x: seq[T]): openarray[T] =
```

Comment:
So if you overrode toOpenArray(openarray) with a new template, this would get called for seq instead of the new override? Why is this here? Same for array override below

Also wouldn't `x.high` evaluate `x` for the second time? That would be confusing. Not to judge but bad PR idea in my opinion